### ISSUE TITLE
from_delayed requires metadata

### ISSUE THREAD

**Issue by jseabold** [dask/dask#1187]
Maybe just a documentation issue? Not clear to me why something like this doesn't work.

```
import pandas as pd
from dask import delayed
import dask.dataframe as dd

XX = [delayed(pd.util.testing.makeDataFrame)() for _ in range(5)]

# doesn't work
df = dd.from_delayed(XX)

# works
df = dd.from_delayed(XX, metadata=['A', 'B', 'C', 'D'])
```

**Comment by mrocklin** [comment]
Agreed.  We should probably also just compute the first delayed value to grab metadata if the user doesn't provide it explicitly.  I can get to this sometime tomorrow afternoon if no one beats me to it.

**Comment by jcrist** [comment]
This should be fixed in #1188. I also clarified the docs - @jseabold does this look good to you?
